src/include/wiredtiger.in

```diff
@@ -1897,6 +1897,24 @@ struct __wt_session {
 	int __F(timestamp_transaction)(WT_SESSION *session, const char *config);
 	/*! @} */
 
+#ifndef DOXYGEN
+	/*!
+	 * Set the commit timestamp on a transaction numerically.  Prefer this over @ref
+	 * timestamp_transaction when the string parsing done in that method becomes a bottleneck.
+	 *
+	 * The WT_SESSION.timestamp_transaction_commit method can only be used at snapshot
```

Comment:
I understand that commit timestamp is the only API where it is causing the performance bottleneck. As we are adding a new API to accepting a numeric timestamp, how about allowing it to support all other timestamps that are supported by the `set_timestamp` function either with an ENUM or MACROS. Wherever the application is possible to send a numeric value they can use it instead of `set_timestamp`?

@sulabhM and @agorrod comments?

Reply:
We did not have an exhaustive discussion about this, but we kind of agree that we are not fond of numeric API, for they will be difficult to change in the future. You cant also pass multiple values to a single call, so the caller will have to make multiple calls to set various timestamps. Hence the proposal is to just get the commit_timestamp API. On the other hand, we should have a larger discussion about these issues.